src/executor/evaluate/mod.rs

```diff
@@ -161,6 +165,45 @@ async fn evaluate_function<'a, T: 'static + Debug>(
 
             Ok(Evaluated::Value(value))
         }
+        name @ "LEFT" | name @ "RIGHT" => {
+            let convert = |string: String, number: usize| {
+                if name == "LEFT" {
+                    string.get(..number)
+                } else {
+                    let start_pos = if number > string.len() {
+                        0
+                    } else {
+                        string.len() - number
+                    };
+                    string.get(start_pos..)
+                }
+                .or(Some(&string))
+                .unwrap()
+                .to_string()
+            };
+
+            if args.len() != 2 {
+                return Err(EvaluateError::NumberOfFunctionParamsNotMatching {
+                    expected: 2,
+                    found: args.len(),
+                }
+                .into());
+            }
+
+            let string = match eval(&args[0]).await?.try_into()? {
+                Value::Str(string) | Value::OptStr(Some(string)) => Ok(string),
+                _ => Err(EvaluateError::FunctionRequiresStringValue(name.to_string())),
+            }?;
+            let number = match eval(&args[1]).await?.try_into()? {
+                Value::I64(number) | Value::OptI64(Some(number)) => usize::try_from(number)
+                    .map_err(|_| EvaluateError::FunctionRequiresUSizeValue(name.to_string())), // Unlikely to occur hence the imperfect error
```

Comment:
This looks good that it can cover not only for the literal but also it can handle `expr`, that's really cool. thanks 👍 
However, current tests only cover literal, `AstValue`.
Can you add these two test cases?
1. A test case which takes `Value::OptI64(Some(number))` branch.
2. An error test case which generates `EvaluateError::FunctionRequiresUSizeValue`